server/base/src/main/java/org/apache/accumulo/server/constraints/MetadataConstraints.java

```diff
@@ -395,4 +266,178 @@ public String getViolationDescription(short violationCode) {
     return null;
   }
 
+  private void validateColValLen(ArrayList<Short> violations, ColumnUpdate columnUpdate) {
+    Text columnFamily = new Text(columnUpdate.getColumnFamily());
+    if (columnUpdate.getValue().length == 0 && !(columnFamily.equals(ScanFileColumnFamily.NAME)
+        || columnFamily.equals(LogColumnFamily.NAME))) {
+      addViolation(violations, 6);
+    }
+  }
+
+  private void validateTabletRow(ArrayList<Short> violations, byte[] row) {
+    // check the row, it should contain at least one ";" or end with "<". Row should also
+    // not be less than AccumuloTable.METADATA.tableId().
+    boolean containsSemiC = false;
+
+    for (byte b : row) {
+      if (b == ';') {
+        containsSemiC = true;
+      }
+
+      if (b == ';' || b == '<') {
+        break;
+      }
+
+      if (!validTableNameChars[0xff & b]) {
+        addIfNotPresent(violations, 4);
+      }
+    }
+
+    if (!containsSemiC) {
+      // see if last row char is <
+      if (row.length == 0 || row[row.length - 1] != '<') {
+        addIfNotPresent(violations, 4);
+      }
+    }
+
+    if (row.length > 0 && row[0] == '!') {
+      if (row.length < 3 || row[1] != '0' || (row[2] != '<' && row[2] != ';')) {
+        addIfNotPresent(violations, 4);
+      }
+    }
+
+    // ensure row is not less than AccumuloTable.METADATA.tableId()
+    if (new Text(row).compareTo(new Text(AccumuloTable.METADATA.tableId().canonical())) < 0) {
+      addViolation(violations, 5);
+    }
+  }
+
+  private void validateTabletFamily(ArrayList<Short> violations, ColumnUpdate columnUpdate,
+      Mutation mutation) {
+    String qualStr = new String(columnUpdate.getColumnQualifier(), UTF_8);
+
+    switch (qualStr) {
+      case (TabletColumnFamily.PREV_ROW_QUAL):
+        if (columnUpdate.getValue().length > 0 && !violations.contains((short) 4)) {
+          KeyExtent ke = KeyExtent.fromMetaRow(new Text(mutation.getRow()));
+
+          Text per = TabletColumnFamily.decodePrevEndRow(new Value(columnUpdate.getValue()));
+
+          boolean prevEndRowLessThanEndRow =
+              per == null || ke.endRow() == null || per.compareTo(ke.endRow()) < 0;
+
+          if (!prevEndRowLessThanEndRow) {
+            addViolation(violations, 3);
+          }
+        }
+        break;
+    }
```

Comment:
Replaced the switch with if. Wrote this under the assumption more cases would be added, but shouldn't assume that, and if more are added this can be swapped with switch. Did not end up including the early returns. Entirely a personal preference, but I like to avoid early returns unless needed, especially in a void method. I think it's easier to read and modify in the future this way.